src/components/AssetTree/AssetTree.tsx

```diff
@@ -76,29 +78,45 @@ class AssetTree extends React.Component<AssetTreeProps, AssetTreeState> {
     super(props);
     const { defaultExpandedKeys } = props;
     this.state = {
+      loading: true,
       rootAssetNodes: [],
       assetNodesMap: {},
       expandedKeys: AssetTree.toKeys(defaultExpandedKeys || []),
     };
   }
 
   async componentDidMount() {
+    this.loadAssetInfo();
+  }
+
+  componentDidUpdate(prevProps: AssetTreeProps) {
+    if (!isEqual(prevProps.assetIds, this.props.assetIds)) {
+      this.loadAssetInfo();
+    }
+  }
+
+  loadAssetInfo = async () => {
     if (!this.context) {
       console.error(ERROR_NO_SDK_CLIENT);
       return;
     }
-    const assets = await this.context.assets
-      .list({ filter: { root: true } })
-      .autoPagingToArray(this.autoPagingToArrayOptions);
-
-    const assetNodesMap = AssetTree.convertAssetsToNodeMap(assets);
-    const rootAssetNodes = Object.keys(assetNodesMap).map(Number);
-
-    this.setState({
-      rootAssetNodes,
-      assetNodesMap,
+    this.setState({ loading: true }, async () => {
+      const { assetIds } = this.props;
+      const assets = await this.context!.assets.list({
+        filter: assetIds ? { parentIds: assetIds } : { root: true },
```

Comment:
`assetIds ? { parentIds: assetIds }` this is wrong? you have to call https://docs.cognite.com/api/v1/#operation/byIdsAssets in this case instead

Reply:
how is this wrong? you only want to show the children of the list of assets you've provided

Reply:
For discovery, for instance, I want to show all assets under IAA in this format, i would pass it the id of IAA.

Reply:
Decided on a new logic!